### PR TITLE
Fixture err return

### PR DESCRIPTION
**Pull request by chcmedeiros**
Replace the return error with setting error. Update some stuff

:link: [zboto Link](https://app.clickup.com/t/34qh19k)

### CODE REVIEW

File: app/src/crypto.c [hunk 1]
```diff
@@ -73,15 +74,15 @@ zxerr_t crypto_extractPublicKey(key_kind_e addressKind, const uint32_t path[HDPA
...
                     CLOSE_TRY;
-                    return zxerr_invalid_crypto_settings;
+                    err = zxerr_invalid_crypto_settings;
             }
         }
         CATCH_ALL
         {
             MEMZERO(&cx_privateKey, sizeof(cx_privateKey));
             MEMZERO(privateKeyData, SK_LEN_25519);
             CLOSE_TRY;
```
Comment (ftheirs) [inline app/src/crypto.c:84]: I think that we should remove this CLOSE_TRY macro

Comment (ftheirs) [reply app/src/crypto.c:84]: we can just clean everything in FINALLY block.. just leave CATCH_ALL empty

File: app/src/crypto.c [hunk 2]
```diff
@@ -224,15 +229,15 @@ zxerr_t crypto_sign_sr25519(uint8_t *signature, uint16_t signatureMaxlen,
         CATCH_ALL
         {
             CLOSE_TRY;
-            return zxerr_unknown;
+            err = zxerr_unknown;
```
Comment (ftheirs) [inline app/src/crypto.c:232]: similar to previous comment

Comment (chcmedeiros) [reply app/src/crypto.c:232]: https://github.com/Zondax/ledger-edgeware/blob/b1a1285f78b5ebe1b81a4471b6424a916f09f750/app/src/crypto.c#L153-L157 
Should we only set the error and *signatureLen = 0 here too @ftheirs ?

Comment (ftheirs) [reply app/src/crypto.c:232]: ohh yeap! Good catch 😄

**Review by ftheirs — APPROVED** [review]

**Pull request merged**